web/src/settings_components.ts

```diff
@@ -483,6 +483,13 @@ export function change_save_button_state($element: JQuery, state: string): void
 
     const $saveBtn = $element.find(".save-button");
     const $textEl = $saveBtn.find(".save-discard-widget-button-text");
+    const $stream_permissions_warning_banner = $(
+        "#stream_permission_settings .stream-permissions-warning-banner",
+    );
+
+    if ($stream_permissions_warning_banner.length > 0) {
+        $stream_permissions_warning_banner.empty();
+    }
```

Comment:
I think it would be better to not add this code here to a function which is used for handling save-discard buttons for all the settings UI. Instead we should just add the code in `discard_property_element_changes` and separately when updating the UI after receiving the event of successful update of stream privacy.

Reply:
Hey @sahil839, I've implemented the changes as per your suggestion. I moved the code into `discard_property_element_changes`, as you advised. Also, `discard_property_element_changes` is already triggered after a successful update event, so there's no need to call it separately. Let me know if you have any further feedback or questions!